index.js

```diff
@@ -303,40 +369,63 @@ Blockchain.prototype._putBlock = function (block, cb, isGenesis) {
 /**
  *Gets a block by its hash
  * @method getBlock
- * @param {String|Buffer|Number} hash - the sha256 hash of the rlp encoding of the block
+ * @param {Buffer|Number|BN} hash - the sha256 hash of the rlp encoding of the block
```

Comment:
Hmm, I'm unsure how to proceed with API documentation. This is a bit of a mess anyhow atm and we should switch to generated ``documentation`` API docs from the code. My tendency is to not update the README on this with this PR and then do the autogeneration on a direct subsequent one and then switch to that and remove the current (already incomplete) API docs from the README.

What do you think?

Reply:
(if you want to take on the extra work you can also just add the ``documentation`` dependency to the dev dependencies, add a npm command like ``"build:docs": "documentation build ./index.js --format md --shallow > ./docs/index.md`` or similar (would be cool to omit the ``_`` functions, not sure if such a flag exists) and the do the documentation changes above)

Reply:
I really like the idea of autogenerating docs. Once this PR is accepted, I'm happy to do another one implementing the approach you describe.

Reply:
This is already done in various of the other ``ethereumjs`` libraries, e.g. in [ethereumjs-block(https://github.com/ethereumjs/ethereumjs-block).